### DISCUSSION TITLE
How is this convention better?

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [sveltejs/kit#5815] [General]
`Instead of defining routes via files and folders, routes are only defined by folders. Layout and page components are defined with +layout.svelte and +page.svelte files — so src/routes/about.svelte would become src/routes/about/+page.svelte.`

Sorry, but I don't understand how this is better, when working on different routes on the editor, we would end with different open files all named `+page.svelte` not making it clear what is what.

Am I missing something here? Because I don't get it. Much rather have `about.svelte` `portfolio.svelte` `photos.svelte` and so on, than having `+page.svelte` `+page.svelte` `+page.svelte`

**Comment by alumpe** [comment]
It is definitely a trade-off, though editors like VSCode show the parent directory in the tab name when you have mutiple files with the same name open.
![](https://i.imgur.com/Pq1a6fU.png)

**Reply by MrAmericanMike** [reply]
This is the problem "When there are multiple files with the same name" but if there is only one, then you don't have a clue. Is there a way to force it to always show the parent directory?

**Reply by eddiemcconkie** [reply]
> Is there a way to force it to always show the parent directory?

There is. Search for "label" in settings and it should be the first one – Workbench > Editor: **Label format**
Set it to "short"

**Reply by lordelix** [reply]
Japp, let's just bully people into using VSCode by enforcing `+page.svelte` files.
Bye bye Notepad++, TextMate, UltraEdit, Coda ... 👋

**Comment by madeleineostoja** [comment]
This was discussed in excruciating detail in the original discussion, there's hundreds of comments there on the rationale and all the arguments for an against. I personally don't love using special characters in the filenames, but directory based routing has several very large benefits, Next.js is moving in the same direction

**Comment by accuser** [comment]
If you haven't already, I'd recommend listening to [Svelte Radio E49](https://www.svelteradio.com/episodes/lets-talk-routing-with-rich-harris). @Rich-Harris explains some of the concerns and nuances that might have been missed in the discussion, and it certainly helped me to understand better the purpose of the changes.

**Reply by MrAmericanMike** [reply]
> If you haven't already, I'd recommend listening to [Svelte Radio E49](https://www.svelteradio.com/episodes/lets-talk-routing-with-rich-harris). @Rich-Harris explains some of the concerns and nuances that might have been missed in the discussion, and it certainly helped me to understand better the purpose of the changes.

Thanks. Listening it right now.

Still looks like Rick had some concepts wrong during the Podcast when he mentions "If it has a plus sign, it's a route file" Which doesn't fit with +layout.svelte that clearly isn't a route.

**Reply by accuser** [reply]
The folder *is* the route. The files with a `+` prefix are route files.
